File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 7/13/2026, 10:17:01 AM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        l=""
        if strs is None or len(strs) == 0:
            return l
        m = len(strs[0])
        for i in range(1, len(strs)):
            m= min(m, len(strs[i]))
        for i in range(0, m):
            for j in range(0, len(strs)):
                if strs[j][i] != strs[0][i]:
                    return l
            l += strs[0][i]
        return l
        

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 7/13/2026, 10:16:44 AM
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String s1 =strs[0];
        String s2 =strs[strs.length-1];
        int len=Math.min(s1.length(),s2.length());
        String s="";
        for(int i=0;i<len;i++){
            if (s1.charAt(i)==s2.charAt(i))
            {
                s=s+s1.charAt(i);
            }
            else 
            {
                break;
            }
        
        }
        return s;
        
    }
}